### PR TITLE
[Zero-Dim] Fix 0-dim tensor for arg_min_max op.

### PR DESCRIPTION
**Pull request by ZHUI**
### PR types

 New features
### PR changes

OPs
### Describe

Fix 0-dim tensor for arg_min_max op.

### CODE REVIEW

File: paddle/phi/infermeta/unary.cc [hunk 1]
```diff
@@ -160,26 +160,39 @@ void ArgMinMaxInferMeta(const MetaTensor& x,
...
+  }
 
-  auto x_rank = x_dims.size();
   if (int_axis < 0) int_axis += x_rank;
+
   if (config.is_runtime) {
     if (dtype == phi::TransToProtoVarType(DataType::INT32)) {
       int64_t all_element_num = 0;
-      if (flatten) {
+      if (x_rank == 0) {
```
Comment (zhwesky2010) [inline paddle/phi/infermeta/unary.cc]: 这个也不用写分支，因为phi::product(x_dims); 里面已经支持了0D的product计算结果是1

File: python/paddle/fluid/tests/unittests/test_zero_dim_tensor.py [hunk 2]
```diff
@@ -46,6 +46,8 @@
     paddle.nn.functional.silu,
     paddle.nn.functional.tanh,
     paddle.nn.functional.dropout,
+    paddle.argmax,
```
Comment (zhwesky2010) [inline python/paddle/fluid/tests/unittests/test_zero_dim_tensor.py]: 这两个应该是reduce系列的，放到下面的reduce_api_list吧

Comment (ZHUI) [reply python/paddle/fluid/tests/unittests/test_zero_dim_tensor.py]: 是 reduce 类似，但是 这个 没有反向梯度。reduc 那边默认会有反向的检查。

File: paddle/phi/infermeta/unary.cc (hunk 1 above)
Comment (ZHUI) [reply paddle/phi/infermeta/unary.cc]: 额，这里有个flatten的配置，怕这个 为 false的话，后面可能有问题。

Comment (zhwesky2010) [reply paddle/phi/infermeta/unary.cc]: 0D的时候axis只能为None，就是flatten的情况

File: python/paddle/fluid/tests/unittests/test_zero_dim_tensor.py (hunk 2 above)
Comment (zhwesky2010) [reply python/paddle/fluid/tests/unittests/test_zero_dim_tensor.py]: 可以给reduce加个不带反向的list，或者放到sundry里单独写吧，因为这个reduce系列后面可能还要加ND->0D的case，elementwise-unary没办法满足

**Review by zhwesky2010 — COMMENTED** [review]
加下axis的检查吧，0D的axis只能为None

File: paddle/phi/infermeta/unary.cc (hunk 1 above)
Comment (ZHUI) [reply paddle/phi/infermeta/unary.cc]: done

File: paddle/phi/kernels/xpu/arg_min_max_kernel.cc [hunk 3]
```diff
@@ -39,7 +40,12 @@ void ArgMaxKernel(const Context& dev_ctx,
           DataType::INT64,
           DataType::INT32,
           dtype));
+  // TODO(ZHUI): fix dtype of out
   dev_ctx.template Alloc<int64_t>(out);
+  if (x.dims().size() == 0) {
+    phi::funcs::set_constant(dev_ctx, out, 0);
```
Comment (zhwesky2010) [inline paddle/phi/kernels/xpu/arg_min_max_kernel.cc]: XPU的设置为常数：

```
xpu::constant<T>(
        dev_ctx.x_context(), dx_data, x->numel(), static_cast<T>(1.0));
```

**Review by zhwesky2010 — COMMENTED** [review]
需要再微调下单测写法：
单测在TestSundry和TestSundryStatic里分别实现，测试axis=0/-1/None三种情况，测试前向shape与前向值

**Review by zhwesky2010 — APPROVED** [review]
